### PR TITLE
Fix builds and ap-airgap inttests on mac m1

### PR DESCRIPTION
**Pull request by juanluisvaladas**
## Description

k0s is unable to build and test on apple silicon. This changes fixes partially some of the issues https://github.com/k0sproject/k0s/issues/2372

WIth this PR the airgap image bundle can be generated correctly and airgap and ap-airgap tests pass.
The scope of tests that remain broken is unknown at this point but at least ap-ha3x3 is still broken.

Also I have snuck a commit to add .DS_Store in the .gitignore, but since that's mac related and very small I think it makes sense

Fixes # (issue)
Partially https://github.com/k0sproject/k0s/issues/2372
## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

- [x] Manual test
- [ ] Auto test added

## Checklist:

- [x] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [x] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -22,3 +22,4 @@ static/zz_generated_assets.go
 docs/cli
 site
 k0sctl.yaml
+.DS_Store
```
Comment (twz123) [inline .gitignore:25]: I'd suggest to put those things into `$XDG_CONFIG_HOME/git/ignore`, so we don't have to accumulate all sorts of OS specific stuff here.

File: Makefile [hunk 2]
```diff
@@ -216,7 +216,7 @@ lint: .k0sbuild.docker-image.k0s go.sum codegen
 	$(GO_ENV) golangci-lint run --verbose $(GOLANGCI_LINT_FLAGS) $(GO_DIRS)
 
 airgap-images.txt: k0s
```
Comment (twz123) [inline Makefile:218]:
```suggestion
airgap-images.txt: k0s .k0sbuild.docker-image.k0s
```

If the Docker container is used to execute the binary, it's better to list it as a make prerequisite.

**Review by twz123 — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (juanluisvaladas) [reply .gitignore]: Makes perfect sense, I removed the commit

File: Makefile (hunk 2 above)
Comment (juanluisvaladas) [reply Makefile]: Done

**Review by twz123 — APPROVED** [review]

**Pull request merged**